components/layout_2020/table/layout.rs

```diff
@@ -689,17 +669,26 @@ impl<'a> TableLayout<'a> {
             .content_box_size(containing_block_for_table, &self.pbm)
             .inline
         {
-            LengthPercentage(_) => resolved_table_width.max(gridmin),
+            LengthPercentage(_) => resolved_table_width.max(grid_min_max.min_content),
             Auto => {
                 let min_width: Au = style
                     .content_min_box_size(containing_block_for_table, &self.pbm)
                     .inline
                     .auto_is(Length::zero)
                     .into();
-                resolved_table_width.clamp(gridmin, gridmax).max(min_width)
+                resolved_table_width
+                    .clamp(grid_min_max.min_content, grid_min_max.max_content)
+                    .max(min_width)
             },
         };
 
+        let border_spacing = self.table.border_spacing();
+        let inline_border_spacing = if self.table.size.width > 0 {
+            border_spacing.inline * (self.table.size.width as i32 + 1)
+        } else {
+            Au::zero()
+        };
+
```

Comment:
Perhaps this could be moved to a helper method that properly returns the total inline and block border spacing in order to put all this logic into the same place.